### ISSUE TITLE
C# 7.2 new feature support: In-parameters causing error: Error extracting metadata for (...) Unable to generate spec reference for !:

### ISSUE THREAD

**Issue by Ollhax** [dotnet/docfx#2385]
**DocFX Version Used**: 
2.30

**Template used**
default

**Steps to Reproduce**:
Sample: [files.zip](https://github.com/dotnet/docfx/files/1667899/files.zip)

Create a new project, set metadata source to a .cs-file with this: 

```
using System;

namespace Namespace
{
	public class Test
	{
		public Foo(int i, in ControlTransform transform)
		{
			
		}
	}
}

```
Run ''docfx metadata'', get the following error:

```
[18-01-26 01:49:47.597]Error:Error extracting metadata for C:/Dev/Projects/Test/DocFX/test.cs: Unable to generate spec reference for !:
        0 Warning(s)
        1 Error(s)
```

If "in" is removed for second parameter in Foo, it works. Also, the first dummy parameter is removed, it works.

**Comment by vicancy** [comment]
Probably should update `Microsoft.CodeAnalysis` dependencies to support the latest c# 7.2.

**Comment by vicancy** [comment]
Should be fixed in v2.31, let me know if there are any other issues.
